static/js/components/google_map.tsx

```diff
@@ -0,0 +1,236 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Component for embedding a map from Google Maps
+ *
+ * Used for plotting a place or location on a map via Google Maps, given that
+ * place, location, or event's DCID. This component can plot either KML
+ * coordinates from the /api/places/mapinfo api as polygons, or a lat/long
+ * marker pin using the node's latitude and longitude properties.
+ *
+ * Note: If a node has both KML coordinates and lat/long properties, the KML
+ * coordinates are preferred.
+ */
+
+import axios from "axios";
+import _ from "lodash";
+import React from "react";
+
+const DEFAULT_MAP_ZOOM = 4;
+
+/**
+ * Response format of /api/place/mapinfo/.
+ */
+interface MapInfoResponse {
+  up: number;
+  down: number;
+  left: number;
+  right: number;
+  coordinateSequenceSet: google.maps.LatLng[][];
+}
+
+/**
+ * Lat/Long coordinates, stored in a form the Google maps API expects.
+ */
+interface GoogleMapCoordinates {
+  lat: number;
+  lng: number;
+}
+
+interface GoogleMapPropType {
+  // DCID of the place/event to show a map for.
+  dcid: string;
+}
+
+interface GoogleMapStateType {
+  // parsed response of propvals api
```

Comment:
nit: it is more useful to indicate what the field actually is/how it is used. i.e., this is the coordinates where a marker should be drawn on the map.

Reply:
thanks for the pointer! Done.